source/agora/consensus/EnrollmentManager.d

```diff
@@ -471,6 +471,37 @@ public class EnrollmentManager
         return this.validator_set.hasPreimage(enroll_key, distance);
     }
 
+    /***************************************************************************
+
+        Generate the random seed reduced from the preimages for the provided
+        block height.
+
+        Params:
+            keys = the keys to look up
+            height = the desired block height to look up the images for
+
+        Returns:
+            the random seed
+
+    ***************************************************************************/
+
+    public Hash getRandomSeed (in Hash[] keys, in Height height) @safe nothrow
+    {
+        assert(keys.length != 0);
```

Comment:
Shouldn't we check if the hash values in the `keys` are sorted in order?

Reply:
Thanks, I'll add the assert.